client/python/openlineage/client/client.py

```diff
@@ -228,6 +228,13 @@ def _http_transport_from_env_variables() -> HttpTransport:
         if endpoint is not None:
             config.endpoint = endpoint
 
+        headers_str = os.environ.get("OPENLINEAGE_HTTP_HEADERS")
```

Comment:
I'm just wondering, if we are later translating this to a dict, is it necessary to introduce another logic here with some splitting and equal signs? Can't we just ask the user to provide json dict in `OPENLINEAGE_HTTP_HEADERS` ? 

Overall, maybe it would be better, instead of adding this feature, to simply implement [#2667](https://github.com/OpenLineage/OpenLineage/issues/2667) - let the user provide transport dict to openlineage client, and in there he could add some extra http headers. I just think that introducing another config variable, that is limited to http transport may not be the best way of handling it in the long run. Let me know your thoughts.

Reply:
Thank you for the review! I think using a dictionary env variable format is less user-friendly than setting a single variable (way easier to make a mistake, you have to set multiple values etc). My goal was to make the onboarding to OpenLineage as simple as possible. 
I'd support both in the long run. Dictionary for consistency / advanced users and single-variable-per-value as a simpler alternative.

Reply:
I understand your perspective. I believe using a dictionary environment variable format can actually enhance user-friendliness, but maybe let's hear some more opinions on this - maybe I'm in a minority 😄 

Also correct me if I'm wrong, but if I tried to provide this custom http headers in a .yml file (and not via env variable) with your current implementation, I'd have to provide it as a dict anyway right? `HttpConfig.from_dict` simply reads the provided parameters but it does not perform the conversion from `key=value` to `{"key": "value"}`. So I think it's not working properly at the moment.

Reply:
Also, one more concern about this format `key=value;key1=value1`. I think both `;` and `=` are valid characters that can be included in HTTP header values (I could not find any confirmation stating otherwise), so providing headers in this way can be erroneous and hard to debug, when it breaks.

Reply:
@kr-igor can we do the one thing without the other? Add http headers to http transport, but not add env variable now, and fix it in https://github.com/OpenLineage/OpenLineage/issues/2916?